### DISCUSSION TITLE
Generate separate resolvers per query

### DISCUSSION THREAD

**Discussion by ddskolla** [kobylynskyi/graphql-java-codegen#789] [Q&A] [answered]
Hi,

Is there any way I could make the generator generate separate resolver interface classes per query ?
Right now, if i have something like this in my schema 

> type Root {
>     allUsers(first: Int): UsersConnection
>     user(id: ID): User

I get a resolver like this 

> public interface RootResolver {
>     UsersConnectionTO allUsers(RootTO rootTO, Integer first) throws Exception;
>     UserTO user(RootTO rootTO, String id) throws Exception;
> }

I would like to generate two seperate resolvers here for the two query methods so that i could make use of the `parentInterfaces` option of the generator to generate resolvers that extend `graphql.schema.DataFetcher` which will allow me to wire in the resolvers directly into the graphql RuntimeWiring class.

Without this, it is really difficult to introduce the data loader concepts in to the mix, feels like the resolvers generated by the generator and the Data fetchers required to create the `GraphQLSchema` try to achieve the same thing, which is why i am trying to connect them together and have just one thing.

Any help would be appreciated.
Thank you.

**Comment by kobylynskyi** [comment, marked as answer]
Hi @ddskolla
By default `apiInterfaceStrategy` is set to `INTERFACE_PER_OPERATION` which means that separate interface classes will be generated for each GraphQL operation.

**Reply by ddskolla** [reply]
thank you.
I had a schema section like below on the .graphqls file which caused all the methods to be generated in one Root resolver file.
Removing the schema section fixed this.

```
schema {
xxxyyy
}
````

**Reply by kobylynskyi** [reply]
Glad it is resolved 🎊
